src/storage/txn/actions/prewrite.rs

```diff
@@ -1417,18 +1442,40 @@ pub mod tests {
             b"v2",
             b"k1",
             &None,
-            10.into(),
+            12.into(),
             false,
             100,
-            10.into(),
+            12.into(),
             1,
             15.into(),
             TimeStamp::default(),
             false,
             kvproto::kvrpcpb::Assertion::None,
             kvproto::kvrpcpb::AssertionLevel::Off,
         );
-        must_locked(&engine, b"k2", 10);
+        must_locked(&engine, b"k2", 12);
+        must_rollback(&engine, b"k2", 12, false);
+
+        // And conflict check is according to the for_update_ts for pessimistic prewrite.
+        // So, it will not report error if for_update_ts is large enough.
+        must_prewrite_put_impl(
```

Comment:
Maybe we should also add some test cases in the `utf` suite to cover the path that pessimistic prewrite happens on index keys, it's a bit complex to tell when to do the check and what's the expected output by now considering async commit protocol and assertion check requirements.
/cc @MyonKeminta @ekexium @zyguan 
What do you think?

Reply:
What's the purpose of the test?